contexts/_internal/useFlooredDecibelDifferenceResult.ts

```diff
@@ -0,0 +1,21 @@
+import { useMemo } from "react";
+import { DecibelDifferenceResult } from "./types";
+import { createValue, isValue, unwrap } from "../../utils/valueOrError";
+
+export const useFlooredDecibelDifferenceResult = (
+  decibelDifferenceResult: DecibelDifferenceResult,
+) => {
+  const flooredDecibelDifferenceResult = useMemo(() => {
+    let left = decibelDifferenceResult.left;
+    let right = decibelDifferenceResult.right;
+    if (isValue(left)) {
+      left = createValue(Math.floor(unwrap(left)));
+    }
+    if (isValue(right)) {
+      right = createValue(Math.floor(unwrap(right)));
+    }
+    return { left, right };
+  }, [decibelDifferenceResult]);
+
+  return { flooredDecibelDifferenceResult };
```

Comment:
If we have a utility function
```ts
export const getFlooredResult = (resultForASingleEar: CalculateDecibelDifferenceResult): CalculateDecibelDifferenceResult => {
    if (isValue(resultForASingleEar)) {
        return createValue(Math.floor(unwrap(resultForASingleEar)));
    }
    return resultForASingleEar;
}
```

Then we can simplify this part.

```suggestion
const flooredDecibelDifferenceResult = useMemo(() => ({
    left: getFlooredResult(decibelDifferenceResult.left),
    right: getFlooredResult(decibelDifferenceResult.right)
}), [decibelDifferenceResult])
```

Reply:
Good suggestion! I've now implemented the changes these changes 💪